lib/NfcoreSchema.groovy

```diff
@@ -0,0 +1,500 @@
+            'config',
+            'd',
+            'D',
+            'dockerize',
+            'h',
+            'log',
+            'q',
+            'quiet',
+            'syslog',
+            'v',
+            'version',
+
+            // Options for `nextflow run` command
+            'ansi',
+            'ansi-log',
+            'bg',
+            'bucket-dir',
+            'c',
+            'cache',
+            'config',
+            'dsl2',
+            'dump-channels',
+            'dump-hashes',
+            'E',
+            'entry',
+            'latest',
+            'lib',
+            'main-script',
+            'N',
+            'name',
+            'offline',
+            'params-file',
+            'pi',
+            'plugins',
+            'poll-interval',
+            'pool-size',
+            'profile',
+            'ps',
+            'qs',
+            'queue-size',
+            'r',
+            'resume',
+            'revision',
+            'stdin',
+            'stub',
+            'stub-run',
+            'test',
+            'w',
+            'with-charliecloud',
+            'with-conda',
+            'with-dag',
+            'with-docker',
+            'with-mpi',
+            'with-notification',
+            'with-podman',
+            'with-report',
+            'with-singularity',
+            'with-timeline',
+            'with-tower',
+            'with-trace',
+            'with-weblog',
+            'without-docker',
+            'without-podman',
+            'work-dir'
+        ]
+        def unexpectedParams = []
+
+        // Collect expected parameters from the schema
+        def expectedParams = []
+        for (group in schemaParams) {
+            for (p in group.value['properties']) {
+                expectedParams.push(p.key)
+            }
+        }
+
+        for (specifiedParam in params.keySet()) {
+            // nextflow params
+            if (nf_params.contains(specifiedParam)) {
+                log.error "ERROR: You used a core Nextflow option with two hyphens: '--${specifiedParam}'. Please resubmit with '-${specifiedParam}'"
+                has_error = true
+            }
+            // unexpected params
+            def params_ignore = params.schema_ignore_params.split(',') + 'schema_ignore_params'
+            def expectedParamsLowerCase = expectedParams.collect{ it.replace("-", "").toLowerCase() }
+            def specifiedParamLowerCase = specifiedParam.replace("-", "").toLowerCase()
+            if (!expectedParams.contains(specifiedParam) && !params_ignore.contains(specifiedParam) && !expectedParamsLowerCase.contains(specifiedParamLowerCase)) {
+                // Temporarily remove camelCase/camel-case params #1035
+                def unexpectedParamsLowerCase = unexpectedParams.collect{ it.replace("-", "").toLowerCase()}
+                if (!unexpectedParamsLowerCase.contains(specifiedParamLowerCase)){
+                    unexpectedParams.push(specifiedParam)
```

Comment:
as you're excluding this from the linting anyway, you could also exchange this file with the newest version from the template (nf-core/tools:dev). Fixed a little bug there.